rails/app/controllers/components_controller.rb

```diff
@@ -4,19 +4,19 @@ class ComponentsController < ApplicationController
   before_action { fetch_component(params[:id]) }
   before_action(only: %i[show]) { current_user.add_recent_workspace(params[:project_id], params[:workspace_id]) }
   before_action(except: %i[show attributes]) { authorize(params[:workspace_object].editors) }
+
   # GET /projects/1/workspaces/1/components/1
   def show # rubocop:disable Metrics/AbcSize
     # Rails.logger.debug params
     component = params[:component_object]
-
-    @component = Kaiju::ComponentJson.as_json(
-      component.id, request.base_url,
-      project_id: params[:project_id], workspace_id: params[:workspace_id], user_id: current_user.id
-    )
+    options = { project_id: params[:project_id], workspace_id: params[:workspace_id], user_id: current_user.id }
 
     respond_to do |format|
-      format.html
-      format.json { render json: JSON.pretty_generate(@component) }
+      format.html { @component = Kaiju::ComponentJson.as_json_lite(component.id, request.base_url, options) }
```

Comment:
What exactly do you need out of this structure? I'm wondering if we could improve performance by delivering only exactly what is required.

Reply:
The only required data I need out of this structure is the url for the component